src/background/custom-filters.js

```diff
@@ -160,6 +161,21 @@ async function updateEngine(text) {
 }
 
 async function update(text, { trustedScriptlets }) {
+  // Add custom content blocks to the end of the list
+  const customContentBlocks = await store.resolve(CustomContentBlocks);
+  let customContentBlocksFilters = 0;
+
+  Object.entries(customContentBlocks.selectors).forEach(
+    ([hostname, selectors]) => {
+      if (selectors) {
+        selectors.forEach((selector) => {
+          text += `\n${hostname}##${selector}`;
+          customContentBlocksFilters++;
+        });
+      }
+    },
+  );
```

Comment:
Bit more readable
```
const filters = Object.entries(customContentBlocks.selectors).reduce(
  (all, [hostname, selectors]) => [
    ...all,
    ...selectors.map((selector) => `${hostname}##${selector}`),
  ],
  []
);

```
or 
```
const filters = [];
for (const [hostname, selectors] of Object.entries(customContentBlocks.selectors)) {
  for (const selector of selectors) {
    filters.append(`${hostname}##${selector}`)
  }
}

text += filters.join(`\n`);
```

Reply:
You've changed the logic, the `selectors` variable can be `null`.